File: voidr-playwright/page_objects/login_page.py

from page_objects.base_page import BasePage

class LoginPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.usernameField = 'input[name="user-name"]'
        self.passwordField = 'input[name="password"]'
        self.loginButton = 'input[name="login-button"]'
        self.errorMessage = 'h3[data-test="error"]'

    def navigate(self, url: str):
        self.page.goto(url)

    def login(self, username: str, password: str):
        self.page.fill(self.usernameField, username)
        self.page.fill(self.passwordField, password)
        self.page.click(self.loginButton)

    def getErrorMessage(self):
        return self.page.text_content(self.errorMessage)

File: playwright-js-voidr/tests/page-objects/login-page.js
// loginPage.js
const BasePage = require('./base-page'); // Certifique-se de que o caminho está correto

class LoginPage extends BasePage {
  constructor(page) {
    super(page);
    this.usernameField = 'input[name="user-name"]';
    this.passwordField = 'input[name="password"]';
    this.loginButton = 'input[name="login-button"]';
    this.errorMessage = 'h3[data-test="error"]';
  }

  async login(username, password) {
    await this.page.fill(this.usernameField, username);
    await this.page.fill(this.passwordField, password);
    await this.page.click(this.loginButton);
  }

  async getErrorMessage() {
    return await this.page.textContent(this.errorMessage);
  }
}

module.exports = LoginPage;
